### ISSUE TITLE
Irregular double setting indexers give incorrect values.

### ISSUE THREAD

**Issue by michaelcheers** [bridgedotnet/Bridge#1291]
### Expected

2

### Actual

1

### Steps To Reproduce

**Example [Live](http://live.bridge.net/#9fc177a89bf3f33aee9cf25eaee46b68)**

```csharp
class TryOut
{
    public int this[int v]
    {
        get
        {
            return 1;
        }
        set
        {
            
        }
    }
}

public class App
{
    [Ready]
    public static void Main()
    {
        var tryOut = new TryOut();
        System.Console.WriteLine(tryOut[0] = 2);
    }
}
```

**Comment by geoffreymcgill** [comment]
Native C# Sample:
https://dotnetfiddle.net/DgVmnP

**Comment by geoffreymcgill** [comment]
Thanks for the report. We have reproduced and will investigate a fix.

**Comment by LeonidVeriga** [comment]
Hi @michaelcheers,

That is an interesting issue.

In your sample, there is no actual setter. Looking at the IL code generated, the value `2` comes from Stack:
```
IL_0000: nop
IL_0001: newobj instance void ClassLibrary9.TryOut::.ctor()
IL_0006: stloc.0
IL_0007: ldloc.0
IL_0008: ldc.i4.0
IL_0009: ldc.i4.2
IL_000a: dup
IL_000b: stloc.1
IL_000c: callvirt instance void ClassLibrary9.TryOut::set_Item(int32, int32)
IL_0011: nop
IL_0012: ldloc.1
IL_0013: call void [mscorlib]System.Console::WriteLine(int32)
IL_0018: nop
IL_0019: ret
```
Bridge.NET generates the same setter (empty). `WriteLine` parameter call is different:
```
console.log((tryOut.setItem(0, 2), tryOut.getItem(0)));
```
I.e. JS code returns "actual" value from the getter that is always `1`.

We can mimic .Net behavior for this construction by implementing the following:
```
var $tmp;
($tmp = 2, tryOut.setItem(0, $tmp), $tmp)
```

However, an empty setter does not look to be useful thing and implementation the apporach above will add more complexity into generated JavaScript.

We will give it more thoughts.

**Comment by michaelcheers** [comment]
@LeonidVeriga, that's why I called it an **irregular** double setter.

**Comment by LeonidVeriga** [comment]
Fixed in #1422
